docs/extensions/network.md

```diff
@@ -0,0 +1,81 @@
+# Gardener Network Extension
+
+Gardener is an open-source project that provides a nested user model. Basically, there are two types of services provided by Gardener to its users:
+
+- Managed: users only request a Kubernetes cluster (Clusters-as-a-Service)
+- Hosted: users utilize Gardener to provide their own managed version of Kubernetes (Cluster-Provisioner-as-a-service)
+
+For the first set of users, the choice of network plugin might not be so important, however, for the second class of users (i.e., Hosted) it is important to be able to customize networking based on their needs.
+
+Furthermore, Gardener provisions clusters on different cloud-providers with different networking requirements. For example, Azure does not support Calico Networking [1], this leads to the introduction of manual exceptions in static add-on charts which is error prune and can lead to failures during upgrades.
+
+Finally, every provider is different, and thus the network always needs to adapt to the infrastructure needs to provider better performance. Consistency does not necessarily lie in the implementation but in the interface.
+
+## Motivation
+
+Prior to the `Network Extensibility` concept, Gardener followed a mono network-plugin support model (i.e., Calico). Although this seemed to be the more stable approach, it did not completely reflect the real use-case.
+The goal of the Gardener Network Extensions is to support different network plugins, therefore, the specification for the network resource won't be fixed and will be customized based on the underlying network plugin.
+
+To do so, a `ProviderConfig` field in the spec will be provided where each plugin will define. Below is an example for how to deploy Calico as the cluster network plugin.
+
+## The Network Extensions Resource
+
+Here is what a typical `Network` resource would look-like:
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: Network
+metadata:
+  name: my-network
+spec:
+  podCIDR: 100.244.0.0/16
+  serviceCIDR: 100.32.0.0/13
+  type: calico
+  providerConfig:
+    apiVersion: calico.networking.extensions.gardener.cloud/v1alpha1
+    backend: bird
+    ipam:
+      cidr: usePodCIDR
+      type: host-local
+    kind: NetworkConfig
+```
+
+The above resources is divided into two parts:
+
+- global configuration (e.g., podCIDR, serviceCIDR, and type)
+- provider specific config (e.g., for calico we can choose to configure a `bird` backend)
```

Comment:
It will be nice if a link to this [docu](https://github.com/gardener/gardener-extensions/blob/master/controllers/networking-calico/docs/usage-as-end-user.md) is added here.